web-common/src/features/canvas/components/charts/variants/CartesianChart.ts

```diff
@@ -202,7 +202,9 @@ export class CartesianChartComponent extends BaseChart<CartesianCanvasChartSpec>
     // Randomly select a measure and dimension if available
     const measures = metricsViewSpec?.measures || [];
     const timeDimension = metricsViewSpec?.timeDimension;
-    const dimensions = metricsViewSpec?.dimensions || [];
+    const dimensions = [...(metricsViewSpec?.dimensions || [])].filter(
+      (d) => d.type === "DIMENSION_TYPE_CATEGORICAL",
```

Comment:
Thoughts on adding a categorical dimension selector to the `MetricsViewSelectors` class?